docs/docsite/rst/community/documentation_contributions.rst

```diff
@@ -109,8 +109,8 @@ Drop the ``--user`` option in the following commands if you use a virtual enviro
 
    .. code-block:: bash
 
-    pip install --user -r docs/docsite/requirements.txt # This file installs the unpinned dependency versions.
-    pip install --user -r tests/requirements.txt # This file installs tested dependency versions.
+    pip install --user -r docs/docsite/requirements.txt # Installs the unpinned dependency versions.
+    pip install --user -r tests/requirements.in -c tests/requirements.txt # Installs tested dependency versions.
```

Comment:
@webknjaz Leaving aside the convenience of the nox/tox approach does this look correct to you? Specifically the `-c tests/requirements.txt` option.

My question stems from the fact that there is a `constraints.in` file and the `-c` option takes a constraints file as a value. Perhaps it's missing some nuance or just wrong but requirements.in and constraints.in generates "requirements.txt" which is actually the full constraints file. Does that make sense?

Would it not achieve the same thing to pip install the generated requirements.txt directly (`-r requirements.txt`)?

Reply:
> Would it not achieve the same thing to pip install the generated requirements.txt directly (`-r requirements.txt`)?

No, because under different runtimes, the actual tree may differ. If there's a dependency that conditionally appears deep in the deptree under Linux and doesn't work under macOS at all, `-r` will try to install it (because you asked it to!) and may crash under macOS while working under Linux. `-c` doesn't mean "install these", it means "if they need to be installed, apply the constraints, no-op otherwise".